shomei/validators.py

```diff
@@ -69,3 +69,25 @@ def validate_github_token(token):
         console.print("[dim]continuing anyway, but double-check if you get auth errors[/dim]\n")
 
     return True, None
+
+def validate_github_username(username):
+    """
+    Validate whether the given username exists on GitHub.
+
+    Args:
+        username: The GitHub username to validate
+
+    Returns:
+        tuple: (exists: bool, error_message: str or None)
+
+    Notes:
+        Print a warning if given username doesn't exist but don't block.
+    """
+    
+    response = requests.get(f"https://api.github.com/users/{username}")
```

Comment:
For an example, here, we should add a timeout here to match the rest of the codebase - all our other API calls use `timeout=10`. something like:

```suggestion
    response = requests.get(f"https://api.github.com/users/{username}", timeout=10)
```

this prevents the request from hanging forever if GitHub's API is slow or not responding

Reply:
Seems a good idea, with a timeout we prevent whether the server doesn't respond, didn't think about it. I commit then!